### ISSUE TITLE
invalid constructor inheritance when extending externs that extend externs

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#7837]
(from #7836)

```haxe
extern class A {
    function new();
}

extern class B extends A {
}

class C extends B {
}

class Main {
	static function main() {
		new C();
	}
}
```

This will currently generate invalid code on some targets (at least js and lua), because `new C()` won't call the super constructor. As @Simn said, constructors on extern classes are not inherited, so it's a problem with extern definition. However in this case, `new C()` should be a compile-time error, since we don't define a constructor on either `B` or `C`.

**Comment by RealyUniqueName** [comment]
Wha't the reason to not inherit constructors on extern classes?

**Comment by RealyUniqueName** [comment]
Duplicated by #9501, fixed by 6f4c2f3
